### PR TITLE
MOBILE-3320 files: Fix clearTmpFiles deleting files that shouldn't be

### PR DESCRIPTION
**Pull request by dpalou**

### CODE REVIEW

File: src/core/features/fileuploader/services/fileuploader.ts [hunk 1]
```diff
@@ -191,14 +191,14 @@ export class CoreFileUploaderProvider {
...
     /**
      * Clear temporary attachments to be uploaded.
-     * Attachments already saved in an offline store will NOT be deleted.
+     * Attachments already saved in an offline store will NOT be deleted, only files in tmp folder will be deleted.
      *
      * @param files List of files.
      */
-    clearTmpFiles(files: (CoreWSFile | FileEntry)[]): void {
-        // Delete the local files.
+    async clearTmpFiles(files: (CoreWSFile | FileEntry)[]): Promise<void> {
```
Comment (NoelDeMartin) [inline src/core/features/fileuploader/services/fileuploader.ts:198]: Why did you convert this to async?

File: src/core/features/fileuploader/services/fileuploader.ts [hunk 2]
```diff
@@ -191,14 +191,14 @@ export class CoreFileUploaderProvider {
...
+        // Delete the temporary files.
         files.forEach((file) => {
-            if ('remove' in file) {
+            if ('remove' in file && CoreFile.removeBasePath(file.toURL()).indexOf(CoreFileProvider.TMPFOLDER) === 0) {
```
Comment (NoelDeMartin) [inline src/core/features/fileuploader/services/fileuploader.ts:201]: You could use `String.startsWith()` method instead.

**Review by NoelDeMartin — APPROVED** [review]

File: src/core/features/fileuploader/services/fileuploader.ts (hunk 1 above)
Comment (dpalou) [reply src/core/features/fileuploader/services/fileuploader.ts]: First I did some changes in the function that required it to be async, but they weren't needed in the end and I forgot to remove it. I just removed it.

File: src/core/features/fileuploader/services/fileuploader.ts (hunk 2 above)
Comment (dpalou) [reply src/core/features/fileuploader/services/fileuploader.ts]: Done!

**Pull request merged**